nano/secure/ledger.cpp

```diff
@@ -209,13 +222,15 @@ void ledger_processor::state_block_impl (nano::state_block const & block_a)
 	result.code = existing ? nano::process_result::old : nano::process_result::progress; // Have we seen this block before? (Unambiguous)
 	if (result.code == nano::process_result::progress)
 	{
-		// Revalidate blocks with epoch links
-		if (!valid_signature || ledger.is_epoch_link (block_a.hashables.link))
+		// Validate block if not verified outside of ledger
+		if (result.verified != nano::signature_verification::valid)
```

Comment:
If `result.verified` is `nano::signature_verification::invalid`, could we return `nano::process_result::bad_signature` without checking the signature again ?

```suggestion
		if (result.verified == nano::signature_verification::invalid)
		{
			result.code = nano::process_result::bad_signature;
		}
		if (result.code == nano::process_result::progress && result.verified != nano::signature_verification::valid)
```

Reply:
This is check if signature_verification is unknown or valid_epoch. Then we need to check it again
Invalid should't go this far